src/qibo/ui/mpldrawer.py

```diff
@@ -0,0 +1,721 @@
+    all_qbits.append(c_qbits + t_qbits)
+
+    flatten_arr = _get_all_tuple_items(all_qbits)
+    return min(flatten_arr), max(flatten_arr)
+
+
+def _get_flipped_index(target, labels):
+    """Get qubit labels from the rest of the line,and return indices
+
+    >>> _get_flipped_index('q0', ['q0', 'q1'])
+    1
+    >>> _get_flipped_index('q1', ['q0', 'q1'])
+    0
+    """
+    nq = len(labels)
+    i = labels.index(target)
+    return nq - i - 1
+
+
+def _rectangle(ax, x1, x2, y1, y2, plot_style):
+    Rectangle = matplotlib.patches.Rectangle
+    x = min(x1, x2)
+    y = min(y1, y2)
+    w = abs(x2 - x1)
+    h = abs(y2 - y1)
+    xm = x + w / 2.0
+    ym = y + h / 2.0
+
+    rect = Rectangle(
+        (x, y),
+        w,
+        h,
+        ec=plot_style["edgecolor"],
+        fc=plot_style["fillcolor"],
+        fill=False,
+        lw=plot_style["linewidth"],
+        label="",
+    )
+    ax.add_patch(rect)
+
+
+def _get_flipped_indices(targets, labels):
+    return [_get_flipped_index(t, labels) for t in targets]
+
+
+def _render_label(label, inits={}):
+    """Slightly more flexible way to render labels.
+
+    >>> _render_label('q0')
+    '$|q0\\\\rangle$'
+    >>> _render_label('q0', {'q0':'0'})
+    '$|0\\\\rangle$'
+    """
+    if label in inits:
+        s = inits[label]
+        if s is None:
+            return ""
+        else:
+            return r"$|%s\rangle$" % inits[label]
+    return r"$|%s\rangle$" % label
+
+
+def _make_cluster_gates(gates_items):
+
+    temp_gates = []
+    temp_mgates = []
+    cluster_gates = []
+
+    for item in gates_items:
+        if len(item) == 2:  # single qubit gates
+            if item[0] == "MEASURE":
+                temp_mgates.append(item)
+            else:
+                if len(temp_gates) > 0:
+                    if item[1] in [tup[1] for tup in temp_gates]:
+                        cluster_gates.append(temp_gates)
+                        temp_gates = []
+                        temp_gates.append(item)
+                    else:
+                        temp_gates.append(item)
+                else:
+                    temp_gates.append(item)
+        else:
+            if len(temp_gates) > 0:
+                cluster_gates.append(temp_gates)
+                temp_gates = []
+
+            if len(temp_mgates) > 0:
+                cluster_gates.append(temp_mgates)
+                temp_mgates = []
+
+            cluster_gates.append([item])
+
+    if len(temp_gates) > 0:
+        cluster_gates.append(temp_gates)
+
+    if len(temp_mgates) > 0:
+        cluster_gates.append(temp_mgates)
+
+    temp_gates = []
+    temp_mgates = []
+
+    return cluster_gates
+
+
+def _build_path(filename):
+    file_path = path.abspath(__file__)  # full path of current file
+    dir_path = path.dirname(file_path)  # full path of the directory from file
+    final_path = path.join(dir_path, filename)  # absolute file path of given file
+    return final_path
+
+
+def _check_list_str(list, str):
```

Comment:
Do not use `list` and `str` as names, they are shadowing builtins (and pretty vague, that is the reason why they are used as names for builtins).

You could use something like:
```suggestion
def _check_list_str(substrings, string):
```
or whatever you prefer.